docs/agent-policy.md

```diff
@@ -0,0 +1,169 @@
+
+
+![Example deployment with shipper](diagrams/components-shipper-example.svg)
+
+This example shows the same scenario, but with the shipper enabled. In this setup, there are shipper components whose job is to send the data upstream, and input components that send their data to a local shipper rather than managing their own independent queue and network connections.
+
+A shipper component also has one output unit and one or more input units, but its input units correspond to the components that write to it rather than to individual data sources.
+
+In this example there are _seven_ components, as Agent has created three shippers to manage the connections to each of the three outputs. The `shipper1` component has two input units, since it receives data from both `filebeat1` and `metricbeat` (with two individual data sources each). All this data is queued by the shipper and forwarded upstream to `elasticsearch1`.
+
+## Agent policy format
+
+The policy is specified in YML, and the basic layout is:
+
+```yml
+outputs:
+  outputName1:
+    type: <output type 1>
+    ...
+  outputName2:
+    type: <output type 2>
+    ...
+inputs:
+  - type: <input type 1>
+    id: <input id 1>
+    ...
+  - type: <input type 2>
+    id: <input id 2>
+    ...
+```
+
+Each output configuration is grouped under the output name, while input configurations appear in a list directly under the root `inputs` field.
+
+For the most part, input and output configurations are passed directly to the underlying components when Agent runs them. However, there are some fields with special meaning or behavior, listed in the following sections. Fields marked "removed" are only used by agent and are removed from the configuration before forwarding it to the client component. Any fields not included below are passed through unchanged.
+
+### Input fields
+
+The following fields within input configurations are handled specially by Agent:
+
+#### `type` (string, required)
+
+The type of the input. This must match the `name` or `aliases` field in component's [input specification](component-specs.md). If the given `type` is an alias, Agent replaces it with its canonical input type (the `name` field in the spec file).
+
+#### `id` (string)
+
+The ID for this input. Each input must have a unique ID, which is used in logging and event metadata. This parameter _should_ be specified, but if it isn't present it defaults to the input type (note that this will be the canonical type, which may be different than the `type` field when using an alias).
+
+#### `use_output` (string, removed)
+
+The output this input should write to. This must match one of the output names from the same policy. Defaults to `default`.
+
+#### `log_level` (string, removed)
+
+The log level for this component. This field is not passed on to the underlying component; instead, Agent implements log level filtering itself. Possible values:
```

Comment:
We do pass the LogLevel to components, but it is per Unit. See Beats' handling of this in https://github.com/elastic/beats/blob/25944e2cdcac19ce214bd0db833e49e046317c42/x-pack/libbeat/management/managerV2.go#L478

Reply:
Yes this is shipped to the unit so it can control is own log level. Elastic Agent sits on top of that, allowing a component to not control its own log level. It is recommend that a component does, otherwise it results in the Elastic Agent doing more processing and removing log lines that the component could have just not provided in the first place.

Reply:
Revisited, I think this section is more accurate now.